### ISSUE TITLE
keepTransformInsideBounds mixes scaled coordinates with unscaled coordinates

### ISSUE THREAD

**Issue by TimoKunze** [anvaka/panzoom#208]
I set bounds to { left: 636, top: 819, right: 636, bottom: 483 }. The parent element has a size of 1064x758. The zoomable element has a size of 1044x1044. With a zoom factor of 1, I can pan as expected, meaning that the boundaries are respected as expected.

Now I zoom out to a scale factor of 0.4. The zoomable element now has a size of 417x417. Now I can pan the zoomable element only within a small rectangle. Behavior seems strange and not natural.

I digged into the panzoom code to find out why this happens. I have not yet found the correct way to fix it, but `keepTransformInsideBounds` seems to be buggy.  It calls `getClientRect` which returns a client rectangle that has been scaled using the current scale factor, and it calls `getBoundingBox` which returns the **unscaled** bounds rectangle. Then it calculates the diff between the scaled client rectangle and the unscaled bounds rectangle. To me this seems wrong.
In a first attempt I tried to scale the bounds rectangle as well. The result was different, without being better.

If I find the correct solution I'll create a pull request. But can you confirm that this is a bug and not intented behavior?

**Comment by anvaka** [comment]
I agree with you it is buggy, and I don't like the API at all. I think what we need is something like `maxBounds` in mapbox: https://docs.mapbox.com/mapbox-gl-js/example/restrict-bounds/

What do you think?

**Comment by TimoKunze** [comment]
Yes, this seems to work a bit better. It takes the scaling into account.
In my case I'm displaying a seating plan on a ticket selling machine that has a FullHD portrait mode (1080x1920) screen. If the seating plan is moved completely outside its parent, it is a bit difficult to bring it back into view because you cannot see where it is. That's why I'm trying to implement some bounds, so that there are always some seats visible.
Today I managed to solve this to some degree by handling the `pan` event and calling `moveTo` if `transform.x` or `transform.y` become out of bounds. Calculating those bounds is a bit difficult though... Currently I also have way too much whitespace around the seats that I cannot remove easily but which disturbs my calculation.

I also allow zoom factors smaller than 1. If the seating plan becomes smaller than the container element, I still allow moving it around. This makes it even more difficult to calculate the bounds. It seemed like a natural feature, but now I'm overthinking it.

So, long story short: I did not manage to fix `keepTransformInsideBounds`. The sample you posted seems like an improvement. And I need to fix my code and also rethink features to make things easier. :)
